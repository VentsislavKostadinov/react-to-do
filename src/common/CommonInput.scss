.input-group {
    max-width: 50%;
    margin: auto;

    .form-label {
        margin: auto;
    }
}

@media only screen and (max-width: 768px) {
  .input-group {
    max-width: 100%;
  }
}
