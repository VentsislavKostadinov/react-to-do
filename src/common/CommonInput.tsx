import { Form, InputGroup } from 'react-bootstrap'
import { CommonInputProps } from '../model/CommonInputProps.types'
import classes from '../style/CommonInput.module.scss'

export const CommonInput = ({
    type,
    label,
    value,
    checked,
    handleChange,
    placeholder,
    className,
    dataTestid,
    children,
}: CommonInputProps) => {
    return (
        <>
            {type === 'checkbox' ? (
                <Form.Check
                    type="checkbox"
                    className="input-checkbox"
                    data-testid="input-checkbox"
                    checked={checked}
                    onChange={handleChange}
                />
            ) : (
                <InputGroup hasValidation className={classes.inputGroup}>
                    <Form.Label>{label}</Form.Label>
                    <Form.Control
                        required
                        checked={checked}
                        className={className}
                        data-testid={dataTestid}
                        type={type}
                        value={value}
                        placeholder={placeholder}
                        onChange={handleChange}
                    />
                    {children}
                </InputGroup>
            )}
        </>
    )
}
