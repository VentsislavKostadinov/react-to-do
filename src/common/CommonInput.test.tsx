import React from 'react'
import { CommonInput } from './CommonInput'
import { CommonInputProps } from '../model/CommonInputProps.types'
import { render, screen, fireEvent } from '@testing-library/react'

describe('CommonInput component', () => {
    const commonInputProps: CommonInputProps = {
        type: 'text',
        label: 'Add text',
        value: 'John Doe',
        checked: false,
        handleChange: jest.fn(),
        placeholder: 'Enter a description',
        dataTestid: 'input-type',
    }

    it('render correctly', () => {
        const { asFragment } = render(<CommonInput {...commonInputProps} />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('type input value', () => {
        const testInputValue = 'John Doe'
        render(<CommonInput {...commonInputProps} />)

        const inputValue = screen.getByTestId('input-type')
        fireEvent.change(inputValue, { target: { value: testInputValue } })

        expect(inputValue).toHaveValue(testInputValue)
    })
})
