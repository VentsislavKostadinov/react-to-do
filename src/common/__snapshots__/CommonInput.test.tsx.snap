// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonInput component render correctly 1`] = `
<DocumentFragment>
  <div
    class="inputGroup input-group has-validation"
  >
    <label
      class="form-label"
    >
      Add text
    </label>
    <input
      class="form-control"
      data-testid="input-type"
      placeholder="Enter a description"
      required=""
      type="text"
      value="John Doe"
    />
  </div>
</DocumentFragment>
`;
